### ISSUE TITLE
Flutter drive with simulator, Error waiting for a debug connection: The log reader failed unexpectedly

### ISSUE THREAD

**Issue by bradyt** [flutter/flutter#77992]
## Steps to Reproduce

1. `flutter channel master`
2. `flutter upgrade`
1. `flutter create counter_app`
2. `cd counter_app`
3. Add changes from https://flutter.dev/docs/cookbook/testing/integration/introduction (diff provided below)
4. `flutter drive --target=test_driver/app.dart --verbose --device-id <id of iphone simulator>`

**Expected results:** 

I expected to see app launch and tests run.

**Actual results:** 

App never launched, instead I see,

```
Error waiting for a debug connection: The log reader failed unexpectedly
```

With `--verbose`, I see

```
Waiting for observatory port to be available...
Error waiting for a debug connection: The log reader failed unexpectedly
```

The issue seems to be with Simulator, driver works fine with Android emulator, a physical iPhone device, or macOS app. Simulator works fine with `flutter run`.

<details>
  <summary>diff</summary>

```diff
4 files changed, 64 insertions(+), 58 deletions(-)
lib/main.dart             | 68 ++++++++---------------------------------------
pubspec.yaml              |  3 ++-
test_driver/app.dart      | 11 ++++++++
test_driver/app_test.dart | 40 ++++++++++++++++++++++++++++

modified   lib/main.dart
@@ -1,28 +1,13 @@
 import 'package:flutter/material.dart';
 
-void main() {
-  runApp(MyApp());
-}
+void main() => runApp(MyApp());
 
 class MyApp extends StatelessWidget {
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
-      ),
-      home: MyHomePage(title: 'Flutter Demo Home Page'),
+      title: 'Counter App',
+      home: MyHomePage(title: 'Counter App Home Page'),
     );
   }
 }
@@ -30,15 +15,6 @@ class MyApp extends StatelessWidget {
 class MyHomePage extends StatefulWidget {
   MyHomePage({Key key, this.title}) : super(key: key);
 
-  // This widget is the home page of your application. It is stateful, meaning
-  // that it has a State object (defined below) that contains fields that affect
-  // how it looks.
-
-  // This class is the configuration for the state. It holds the values (in this
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
-
   final String title;
 
   @override
@@ -50,47 +26,18 @@ class _MyHomePageState extends State<MyHomePage> {
 
   void _incrementCounter() {
     setState(() {
-      // This call to setState tells the Flutter framework that something has
-      // changed in this State, which causes it to rerun the build method below
-      // so that the display can reflect the updated values. If we changed
-      // _counter without calling setState(), then the build method would not be
-      // called again, and so nothing would appear to happen.
       _counter++;
     });
   }
 
   @override
   Widget build(BuildContext context) {
-    // This method is rerun every time setState is called, for instance as done
-    // by the _incrementCounter method above.
-    //
-    // The Flutter framework has been optimized to make rerunning build methods
-    // fast, so that you can just rebuild anything that needs updating rather
-    // than having to individually change instances of widgets.
     return Scaffold(
       appBar: AppBar(
-        // Here we take the value from the MyHomePage object that was created by
-        // the App.build method, and use it to set our appbar title.
         title: Text(widget.title),
       ),
       body: Center(
-        // Center is a layout widget. It takes a single child and positions it
-        // in the middle of the parent.
         child: Column(
-          // Column is also a layout widget. It takes a list of children and
-          // arranges them vertically. By default, it sizes itself to fit its
-          // children horizontally, and tries to be as tall as its parent.
-          //
-          // Invoke "debug painting" (press "p" in the console, choose the
-          // "Toggle Debug Paint" action from the Flutter Inspector in Android
-          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
-          // to see the wireframe for each widget.
-          //
-          // Column has various properties to control how it sizes itself and
-          // how it positions its children. Here we use mainAxisAlignment to
-          // center the children vertically; the main axis here is the vertical
-          // axis because Columns are vertical (the cross axis would be
-          // horizontal).
           mainAxisAlignment: MainAxisAlignment.center,
           children: <Widget>[
             Text(
@@ -98,16 +45,23 @@ class _MyHomePageState extends State<MyHomePage> {
             ),
             Text(
               '$_counter',
+              // Provide a Key to this specific Text widget. This allows
+              // identifying the widget from inside the test suite,
+              // and reading the text.
+              key: Key('counter'),
               style: Theme.of(context).textTheme.headline4,
             ),
           ],
         ),
       ),
       floatingActionButton: FloatingActionButton(
+        // Provide a Key to this button. This allows finding this
+        // specific button inside the test suite, and tapping it.
+        key: Key('increment'),
         onPressed: _incrementCounter,
         tooltip: 'Increment',
         child: Icon(Icons.add),
-      ), // This trailing comma makes auto-formatting nicer for build methods.
+      ),
     );
   }
 }
modified   pubspec.yaml
@@ -30,8 +30,9 @@ dependencies:
   cupertino_icons: ^1.0.2
 
 dev_dependencies:
-  flutter_test:
+  flutter_driver:
     sdk: flutter
+  test: any
 
 # For information on the generic Dart part of this file, see the
 # following page: https://dart.dev/tools/pub/pubspec
new file   test_driver/app.dart
@@ -0,0 +1,11 @@
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:counter_app/main.dart' as app;
+
+void main() {
+  // This line enables the extension.
+  enableFlutterDriverExtension();
+
+  // Call the `main()` function of the app, or call `runApp` with
+  // any widget you are interested in testing.
+  app.main();
+}
new file   test_driver/app_test.dart
@@ -0,0 +1,40 @@
+// Imports the Flutter Driver API.
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Counter App', () {
+    // First, define the Finders and use them to locate widgets from the
+    // test suite. Note: the Strings provided to the `byValueKey` method must
+    // be the same as the Strings we used for the Keys in step 1.
+    final counterTextFinder = find.byValueKey('counter');
+    final buttonFinder = find.byValueKey('increment');
+
+    FlutterDriver driver;
+
+    // Connect to the Flutter driver before running any tests.
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+    });
+
+    // Close the connection to the driver after the tests have completed.
+    tearDownAll(() async {
+      if (driver != null) {
+        driver.close();
+      }
+    });
+
+    test('starts at 0', () async {
+      // Use the `driver.getText` method to verify the counter starts at 0.
+      expect(await driver.getText(counterTextFinder), "0");
+    });
+
+    test('increments the counter', () async {
+      // First, tap the button.
+      await driver.tap(buttonFinder);
+
+      // Then, verify the counter text is incremented by 1.
+      expect(await driver.getText(counterTextFinder), "1");
+    });
+  });
+}
```

</details>
<details>
  <summary>flutter analyze</summary>

```
% flutter analyze                                                             
Analyzing counter_app...                                                
No issues found! (ran in 12.2s)
```

</details>
<details>
  <summary>flutter doctor -v</summary>

```
% flutter doctor -v
[✓] Flutter (Channel master, 2.1.0-11.0.pre.224, on Mac OS X 10.15.7 19H524 darwin-x64, locale en-US)
    • Flutter version 2.1.0-11.0.pre.224 at /Users/alice/opt/flutter-master
    • Framework revision 95d3ac703f (3 hours ago), 2021-03-11 17:22:14 -0800
    • Engine revision 1b68503bc8
    • Dart version 2.13.0 (build 2.13.0-125.0.dev)

...

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

...

[✓] Connected device (4 available)
    • sdk gphone x86 arm (mobile) • emulator-5554                            • android-x86 • Android 11 (API 30) (emulator)
    • <id> (mobile)               • <id>                                     • ios         • iOS 14.4.1
    • iPhone 12 Pro Max (mobile)  • 12298D0B-47E5-491D-BF10-A66EE1459380     • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)             • macos                                    • darwin-x64  • Mac OS X 10.15.7 19H524 darwin-x64

! Doctor found issues in 1 category.
```

</details>

**Comment by bradyt** [comment]
TL;DR: To fix, quit and reopen Simulator.

Hmm, I think I fixed the issue as it is described above. Amongst the things I had tried, and terms I had searched internet for, I just tried a moment ago for a search for "xcrun simctl launch flutter", which was in my logs a few lines up. From the third search result, a Stack Overflow answer for a different problem, had one user report that restarting the Simulator fixed it for them. I had already tried a restart from the app's menu, but this time I did a full quit and re-launch, and the previous error as described above is no longer there.

I suppose I should close this, will just explore my issue for a bit longer, checking the new error I'm seeing, and checking if my issue is not just fixed in this minimal broken example.

**Comment by bradyt** [comment]
Hmm, I missed a step above. Before I can get that same error, I first see this error:

<details>
  <summary>first error</summary>

```
% flutter drive --target=test_driver/app.dart --device-id iphone
Running "flutter pub get" in counter_app...                        838ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                      2,289ms
Xcode build done.                                            6.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    building file list ... done

    sent 544 bytes  received 20 bytes  1128.00 bytes/sec
    total size is 43092281  speedup is 76404.75
    building file list ... done

    sent 630 bytes  received 20 bytes  1300.00 bytes/sec
    total size is 38734939  speedup is 59592.21
    error: Runner.app/Info.plist does not exist. The Flutter "Thin Binary" build phase must run after "Copy Bundle Resources".
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Application failed to start on attempt: 1
Application failed to start on attempt: 2
Application failed to start on attempt: 3
Application failed to start. Will not run test. Quitting.
```

</details>

I then try `open ios/Runner.xcworkspace`, apply the changes that are suggested, which produces this diff:


<details>
  <summary>changes from Xcode</summary>

```diff
2 files changed, 25 insertions(+), 16 deletions(-)
ios/Runner.xcodeproj/project.pbxproj               | 31 +++++++++++++++-------
.../xcshareddata/xcschemes/Runner.xcscheme         | 10 +++----

modified   ios/Runner.xcodeproj/project.pbxproj
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 46;
+	objectVersion = 50;
 	objects = {
 
 /* Begin PBXBuildFile section */
@@ -127,7 +127,7 @@
 		97C146E61CF9000F007C117D /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 1020;
+				LastUpgradeCheck = 1240;
 				ORGANIZATIONNAME = "";
 				TargetAttributes = {
 					97C146ED1CF9000F007C117D = {
@@ -254,6 +254,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
 				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
 				CLANG_WARN_STRICT_PROTOTYPES = YES;
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
@@ -272,7 +273,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
 				SDKROOT = iphoneos;
 				SUPPORTED_PLATFORMS = iphoneos;
@@ -290,7 +291,10 @@
 				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
 				ENABLE_BITCODE = NO;
 				INFOPLIST_FILE = Runner/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.example.counterApp;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
@@ -322,6 +326,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
 				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
 				CLANG_WARN_STRICT_PROTOTYPES = YES;
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
@@ -346,7 +351,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
 				SDKROOT = iphoneos;
@@ -377,6 +382,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
 				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
 				CLANG_WARN_STRICT_PROTOTYPES = YES;
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
@@ -395,11 +401,12 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
 				SDKROOT = iphoneos;
 				SUPPORTED_PLATFORMS = iphoneos;
-				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
+				SWIFT_COMPILATION_MODE = wholemodule;
+				SWIFT_OPTIMIZATION_LEVEL = "-O";
 				TARGETED_DEVICE_FAMILY = "1,2";
 				VALIDATE_PRODUCT = YES;
 			};
@@ -414,7 +421,10 @@
 				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
 				ENABLE_BITCODE = NO;
 				INFOPLIST_FILE = Runner/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.example.counterApp;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
@@ -433,7 +443,10 @@
 				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
 				ENABLE_BITCODE = NO;
 				INFOPLIST_FILE = Runner/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.example.counterApp;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
modified   ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1020"
+   LastUpgradeVersion = "1240"
    version = "1.3">
    <BuildAction
       parallelizeBuildables = "YES"
@@ -27,8 +27,6 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
-      <Testables>
-      </Testables>
       <MacroExpansion>
          <BuildableReference
             BuildableIdentifier = "primary"
@@ -38,8 +36,8 @@
             ReferencedContainer = "container:Runner.xcodeproj">
          </BuildableReference>
       </MacroExpansion>
-      <AdditionalOptions>
-      </AdditionalOptions>
+      <Testables>
+      </Testables>
    </TestAction>
    <LaunchAction
       buildConfiguration = "Debug"
@@ -61,8 +59,6 @@
             ReferencedContainer = "container:Runner.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
-      <AdditionalOptions>
-      </AdditionalOptions>
    </LaunchAction>
    <ProfileAction
       buildConfiguration = "Profile"
```

</details>

Then I try `flutter clean`, and I get back to the error I described in original issue.

**Comment by bradyt** [comment]
I wonder if this is related to https://github.com/flutter/flutter/issues/70422.

Anyhow, underlying issue is, how were integration tests, automated screenshots, etc, bumped off of stable. 😳

**Comment by darshankawar** [comment]
@bradyt 
`flutter driver` has now been deprecated in favor of [integration_test](https://pub.dev/packages/integration_test) plugin. Can you follow the steps mentioned in the plugin readme and run it on simulator to see if you still get same error or not ?

**Comment by darshankawar** [comment]
@bradyt 
Tried the official `integration_test` [plugin](https://pub.dev/packages/integration_test/install) using latest master and ran it on iOS simulator properly without any issues.

```
Running pod install...                                           1,302ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         7.1s
Xcode build done.                                           24.2s
VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:58410/S4EL8xZsOIw=/
VMServiceFlutterDriver: Isolate found with number: 3491962721731795
VMServiceFlutterDriver: Isolate is paused at start.
VMServiceFlutterDriver: Attempting to resume isolate
flutter: 00:00 +0: failing test example
VMServiceFlutterDriver: Connected to Flutter application.
flutter: 00:00 +1: (tearDownAll)
flutter: 00:00 +2: All tests passed!
All tests passed.
```
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.1.0-11.0.pre.227, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.1.0-11.0.pre.227 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 66768f8c5b (10 hours ago), 2021-03-11 19:55:02 -0800
    • Engine revision 1b68503bc8
    • Dart version 2.13.0 (build 2.13.0-125.0.dev)

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 89.0.4389.82

! Doctor found issues in 1 category.

```

</details>

I used the default command mentioned in readme to run. Did you use the same or the one you shared earlier ?

**Comment by jonahwilliams** [comment]
@darshankawar flutter drive is definitely not deprecated

**Comment by bradyt** [comment]
> Tried the official integration_test plugin using latest master and ran it on iOS simulator properly without any issues.

That is in some ways reassuring. Thank you for testing. I guess now it's just a matter of figuring out the difference. Did you save your project state locally, in case we look for any differences at a later date? I can create my minimal broken example shortly, and push to a public repo.

I guess some possible candidates are, the exact state of app project, the versions of Flutter, Xcode, CocoaPods, iOS, etc. Are you ignoring the CocoaPods recommendation for any particular reason?

> I used the default command mentioned in readme to run. Did you use the same or the one you shared earlier ?

The same.

**Comment by bradyt** [comment]
/me added the `waiting for merchant response` label. 🤪

**Comment by bradyt** [comment]
Here is a repo to share, with explicit commits as steps: https://github.com/bradyt/counter_app_integration_test/tree/cf9a77b0e3f9b67e6c8a70c6bbeafd427c153ab0.

That includes flutter create, adding changes from integration_test README, and running flutter drive, recorded in the Makefile.

**Comment by bradyt** [comment]
```
% make           
flutter drive \
	--driver=test_driver/integration_test.dart \
	--target=integration_test/foo_test.dart
Running "flutter pub get" in counter_app...                        616ms
Running pod install...                                           1,220ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         4.6s
Xcode build done.                                           21.0s
Error waiting for a debug connection: The log reader failed unexpectedly
Application failed to start on attempt: 1
Error waiting for a debug connection: The log reader failed unexpectedly
Application failed to start on attempt: 2
Error waiting for a debug connection: The log reader failed unexpectedly
Application failed to start on attempt: 3
Application failed to start. Will not run test. Quitting.
make: *** [drive] Error 1
```

**Comment by bradyt** [comment]
Hmm, restarting my computer has completely changed things.

I now get,

```
VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:49381/GWwCsFzAGGo=/
VMServiceFlutterDriver: Isolate found with number: 1732512452243775
```

Instead of,

```
Error waiting for a debug connection: The log reader failed unexpectedly
```

This issue is, in some sense, fixed for me. If anyone sees it again, perhaps there are troubleshooting methods to identify what's going on, or a more surgical fix than, "have you tried turning it off and on again?".

I'll defer to Flutter team if and when to close issue. Let me know if there's anything I can do to investigate further.

**Comment by bradyt** [comment]
@darshankawar Thanks again for the "works fine here", definitely pointed me towards improving my reproduce, which finally did not.

**Comment by darshankawar** [comment]
> @darshankawar flutter drive is definitely not deprecated

@jonahwilliams  Hi, thanks for this info. I've noted this and will keep in mind henceforth.

**Comment by darshankawar** [comment]
@bradyt 
Assuming your original issue is resolved, I am going ahead and closing this. If you disagree or see similar issue again, write in comments and I'll reopen it.
Thanks.
